packages/react-magma-dom/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,321 @@
+    let [hideLabelsLayout, setHideLabelsLayout] = React.useState(false);
+    let [summaryViewLayout, setSummaryViewLayout] = React.useState(false);
+
+    const [windowWidth, setWindowWidth] = React.useState(window.innerWidth);
+
+    React.useEffect(() => {
+      function handleResize() {
+        setWindowWidth(window.innerWidth);
+        if (window.innerWidth < breakpoint && breakpoint) {
+          if (layout === StepperLayout.showLabels) {
+            setShowLabelsLayout(false);
+          } else if (layout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(false);
+          } else if (layout === StepperLayout.summaryView) {
+            setSummaryViewLayout(false);
+          }
+          if (breakpointLayout === StepperLayout.showLabels) {
+            setShowLabelsLayout(true);
+          } else if (breakpointLayout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(true);
+          } else if (breakpointLayout === StepperLayout.summaryView) {
+            setSummaryViewLayout(true);
+          }
+        }
+        if (window.innerWidth > breakpoint || !breakpoint) {
+          if (breakpointLayout === StepperLayout.showLabels) {
+            setShowLabelsLayout(false);
+          } else if (breakpointLayout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(false);
+          } else if (breakpointLayout === StepperLayout.summaryView) {
+            setSummaryViewLayout(false);
+          }
+          if (layout === StepperLayout.showLabels) {
+            setShowLabelsLayout(true);
+          } else if (layout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(true);
+          } else if (layout === StepperLayout.summaryView) {
+            setSummaryViewLayout(true);
+          }
+        }
+        if (showLabelsLayout) {
+          setHideLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (hideLabelsLayout) {
+          setShowLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (summaryViewLayout) {
+          setShowLabelsLayout(false);
+          setHideLabelsLayout(false);
+        }
+      }
+
+      window.addEventListener('resize', handleResize);
+      handleResize();
+
+      return () => window.removeEventListener('resize', handleResize);
+    }, [windowWidth]);
+
+    // Step states
+    const step = React.Children.toArray(children);
+
+    const Steps = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        if (child.type === Step) {
+          const stepStatusStyles =
+            currentStep >= index + 1
+              ? StepStatus.complete
+              : currentStep >= index
+              ? StepStatus.active
+              : StepStatus.incomplete;
+
+          const item = React.cloneElement(child, {
+            'aria-current': currentStep === index,
```

Comment:
From a quick look it's unclear to be whether this should evaluate to true/false or 'step'/false (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) but we should investigate. I have seen examples that do both.